src/tests/comparison_tests/test_s21_is_less.c

```diff
@@ -1,47 +1,56 @@
 #include "../s21_decimal_test.h"
 
 START_TEST(gmp_random) {
-    s21_decimal a1 = {0};
-    init_zero(&a1);
-
+    s21_decimal dec1 = {0};
     mpz_t mpz_val, mpz_copy;
-    mpz_init(mpz_val);
-    mpz_init(mpz_copy);
-    mpz_set_ui(mpz_val, 0);
-    mpz_set_ui(mpz_copy, 0);
 
-    get_random_pair(&a1, &mpz_val, 3);
+    mpz_init_set_ui(mpz_val, 0);
+    mpz_init_set_ui(mpz_copy, 0);
 
-    convert_decimal_to_gmp(&mpz_copy, &a1);
-    s21_decimal res = convert_gmp_to_decimal(mpz_val);
+    get_random_pair(&dec1, &mpz_val, 3);
 
-    /* Produces randomly smaller / greater numbers */
-    if (rand() % 2)
-        set_exponent(&a1, (get_exponent(a1) + 1) % 28);
+    convert_decimal_to_gmp(&mpz_copy, &dec1);
+    s21_decimal dec2 = convert_gmp_to_decimal(mpz_val);
 
-    /* (!) Hack. Change this to our own function */
-    tmp_normalize_exponent(&a1);
+    /* d1 is always < d2 */
+    gmp_sum_int(&mpz_copy, 100);
 
     // Compare op1 and op2. Return a positive value if op1 > op2, zero if op1 = op2, or a negative value if op1 < op2.
-    int got = (s21_is_less(a1, res));
-    int expected = mpz_cmp(mpz_val, mpz_copy);
+    int got = s21_is_less(dec1, dec2);
 
-    mpz_clear(mpz_copy);
-    mpz_clear(mpz_val);
+    /* BUG (!) MPZ IS UNABLE TO CORRECTLY COMPARE UNSIGNED INTS */
+    int expected = mpz_cmp(mpz_val, mpz_copy);
+    /* BUG (!) I NEED TO WRITE MY OWN LOGIC!!!! */
 
-    if (expected < 0) {
-        ck_assert_int_eq(got, expected);
+    if (got == 1 && expected < 0) {
+        ck_assert_int_eq(got, 1);
     } else {
-        ck_assert_int_eq(got, 0);
+        printf("GOT: %d EXPECTED: %d\n", got, expected);
+        print_bits_r(dec1);
+        print_bits_r(dec2);
+        printf("MPZ: \n");
+        
+        print_mpz_decimal(mpz_val);
+        print_mpz_decimal(mpz_copy);
+
+        printf("END: \n");
+        ck_assert_int_eq(1, 0);
     }
+    
+    mpz_clear(mpz_copy);
+    mpz_clear(mpz_val);
 }
 END_TEST
 
-Suite *suite_s21_is_less(void) {
+// A : 1959779897 
+// B : 580803652
+
+Suite *
+suite_s21_is_less(void) {
     Suite *s = suite_create(PRETTY_PRINT("s21_is_less"));
     TCase *tc = tcase_create("s21_is_less_tc");
 
-    tcase_add_loop_test(tc, gmp_random, 0, 100);
```

Comment:
As I said, comparisons are generally really stable. They DO indeed work with all vanilla / hardcoded s21_decimals. I just really want to have fancy GMP tests. The only problem with comparisons is that they rely on division, which relies on them. Maybe there is a way to compare numbers without normalizing exponent.